src/test/java/teammates/test/cases/BaseComponentTestCase.java

```diff
@@ -97,7 +97,8 @@ protected StudentAttributes getStudent(StudentAttributes student) {
     }
 
     protected void removeAndRestoreTypicalDataBundle() {
-        removeAndRestoreDataBundle(typicalBundle);
+        DataBundle dataBundle = getTypicalDataBundle();
+        removeAndRestoreDataBundle(dataBundle);
```

Comment:
`DataBundle dataBundle = getTypicalDataBundle(); removeAndRestoreDataBundle(tdataBundle);`

In this we are updating the DB with new DataBundle. And its not same as 
`removeAndRestoreDataBundle(typicalBundle);` as `typicalBundle` which is modified in between
test.